src/components/mermaidMap/MiniMap/MiniMap.js

```diff
@@ -0,0 +1,137 @@
+  border: 2px solid white;
+  opacity: ${({ visible }) => (visible ? 1 : 0)};
+`
+
+const MiniMap = ({ mainMap }) => {
+  const miniMapContainer = useRef(null)
+  const miniMap = useRef(null)
+  const trackingRectSource = useRef(null)
+  const defaultZoom = 2
+  const zoomAdjustment = 5
+
+  const getIsMiniMapVisible = () => mainMap.getZoom() > defaultZoom
+
+  const [isVisible, setIsVisible] = useState(getIsMiniMapVisible())
+
+  const updateTrackingRectGeometry = (bounds) => {
+    if (trackingRectSource.current) {
+      trackingRectSource.current.setData({
+        type: 'Feature',
+        properties: {},
+        geometry: {
+          type: 'Polygon',
+          coordinates: [
+            [
+              [bounds.getWest(), bounds.getNorth()],
+              [bounds.getEast(), bounds.getNorth()],
+              [bounds.getEast(), bounds.getSouth()],
+              [bounds.getWest(), bounds.getSouth()],
+              [bounds.getWest(), bounds.getNorth()],
+            ],
+          ],
+        },
+      })
+    }
+  }
+
+  const addTrackingRectLayers = () => {
+    miniMap.current.addLayer({
+      id: 'trackingRectOutline',
+      type: 'line',
+      source: 'trackingRect',
+      paint: {
+        'line-color': '#FF0000',
+        'line-width': 2,
+      },
+    })
+
+    miniMap.current.addLayer({
+      id: 'trackingRectFill',
+      type: 'fill',
+      source: 'trackingRect',
+      paint: {
+        'fill-color': '#FF0000',
+        'fill-opacity': 0.3,
+      },
+    })
+  }
+
+  const onMapLoad = () => {
+    miniMap.current.addSource('trackingRect', {
+      type: 'geojson',
+      data: {
+        type: 'Feature',
+        properties: {},
+        geometry: { type: 'Polygon', coordinates: [[]] },
+      },
+    })
+
+    setIsVisible(getIsMiniMapVisible())
+
+    addTrackingRectLayers()
+
+    miniMap.current.jumpTo({
+      center: mainMap.getCenter(),
+      zoom: mainMap.getZoom() - zoomAdjustment,
+    })
+
+    trackingRectSource.current = miniMap.current.getSource('trackingRect')
+    updateTrackingRectGeometry(mainMap.getBounds())
+  }
+
+  const initializeMap = () => {
+    miniMap.current = new maplibregl.Map({
+      container: miniMapContainer.current,
+      style: lightBaseMap,
+      center: mainMap.getCenter(),
+      zoom: defaultZoom,
+      interactive: false,
+    })
+
+    miniMap.current.on('load', onMapLoad)
+  }
+
+  const handleMapMove = () => {
+    miniMap.current.jumpTo({
+      center: mainMap.getCenter(),
+      zoom: mainMap.getZoom() - zoomAdjustment,
+    })
+
+    setIsVisible(getIsMiniMapVisible())
+    updateTrackingRectGeometry(mainMap.getBounds())
+  }
+
+  useEffect(() => {
+    if (!miniMapContainer.current || !mainMap) {
+      return
+    }
+
+    initializeMap()
+    mainMap.on('move', handleMapMove)
+
+    // eslint-disable-next-line consistent-return
+    return () => miniMap.current.remove()
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mainMap])
```

Comment:
intentionally not including functions in the dependencies list

Reply:
this would require moving most the functions within the `useEffect`, which I had set up before. I can move it back this way

Reply:
dependency arrays are what make me curious about alternative frameworks in general FWIW :)